### DISCUSSION TITLE
OpenTelemetry error - Additional logs

### DISCUSSION THREAD

**Discussion by jasonddown** [mburumaxwell/paklo#1355] [Q&A] [answered]
I've set up dependabot-azure-devops a couple of weeks ago and making small tweaks since then.
It's been two days now that the run fails with this error message in the pipeline logs:
`ERROR -- : OpenTelemetry error: Unable to export 24 spans`
`ERROR -- : OpenTelemetry error: Unable to export 16 spans`

It seems to happen at the same step, just after:
`INFO <job_1727169139> Updating eslint from 7.32.0 to 9.11.1`

Is there a way to get additional internal logs of dependabot-azure-devops?
Or any ideas what may be causing the error?

**Comment by mburumaxwell** [comment]
You can safely ignore this. It is meant to be used with Sentry but it also requires other config to be setup which I haven;t gotten to just yet.

**Reply by jasonddown** [reply]
The process gets stuck though after getting those error messages, nothing happens until the pipeline times out

**Reply by rhyskoedijk** [reply]
there is likely another error somewhere in the logs as the OpenTelemetry errors shouldn't impact the task success/fail result.
you can set the pipeline variable `System.Debug = true` _(note that "true" must be lower-case, fixed in [1.35.0](https://github.com/tinglesoftware/dependabot-azure-devops/releases/tag/1.35.0))_ to get more verbose logging, but I would expect that the error is already in the logs even without this.

would you be able to share more of the task log around the start of the update?

**Reply by jasonddown** [reply]
Absolutely, here's the logs for the whole Task. 
I did manually cancel this run because as you said, I had tried it with `System.Debug=true`, but the previous runs timed out after the 6 hours pipeline timeout without any further lines logged.
[dependabot-pipeline.log](https://github.com/user-attachments/files/17145492/dependabot-pipeline.log)

**Reply by rhyskoedijk** [reply]
Surprisingly there is no extra information in that log that gives me any ideas as to what was stalled that package update, which is strange...

Would you be in a position to try out the new preview [Task V2](https://github.com/tinglesoftware/dependabot-azure-devops/blob/main/docs/migrations/v1-to-v2.md#summary-of-changes-v1--v2)? When `System.Debug = true`, V2 should dump more verbose logging during the package update, notably from the "proxy" component. I would hope that either V2 doesn't stall or at least provides more useful logs to diagnose the issue.

**Reply by jasonddown** [reply]
Thanks, I'll try that, although I noticed the limitation with the `open-pull-requests-limit: 0` which will not make it a like for like comparison because I am currently using it and will have to change it, but will give it a go

**Reply by rhyskoedijk** [reply]
> I noticed the limitation with the `open-pull-requests-limit: 0` which will not make it a like for like comparison because I am currently using it

Ah, no worries then. I'm currently working on adding security-only updates for Task V2, it shouldn't be too far away. It might be best to wait until that is ready before trying V2 to avoid changing your workflow. 

I did have a fresh look over the logs again but still don't see any clues as to why it's taking so long to update eslint.
Is it worth manually updating eslint using npm while you wait for a fix? If Dependabot doesn't get caught up by this package, would it run to completion or are there other packages after eslint that it also hangs on?

**Reply by jasonddown** [reply]
I've run it with V2 but unfortunately it still hangs and doesn't seem to log much more.  
There's only a couple more lines in V2 after the updater. 
Apart from a bit more debugging info I can't see any major differences in the log between V1 and V2 in general, but I can sanitise the complete task's log and upload t if you are interested?
Below are the last few lines with the couple additional lines compared to V1.
Like you said, I'll update eslint manually and see if it gets it unstuck now unless you have any other ideas that you'd like to try before I do. 

> updater | 2024/10/03 12:06:14 INFO <job_update_0_npm_and_yarn_all> Latest version is 9.11.1
  proxy | 2024/10/03 12:06:17 [264] GET [https://registry.npmjs.org:443/@angular-eslint%2feslint-plugin](https://registry.npmjs.org/@angular-eslint%2feslint-plugin)
  proxy | 2024/10/03 12:06:17 [264] 200 [https://registry.npmjs.org:443/@angular-eslint%2feslint-plugin](https://registry.npmjs.org/@angular-eslint%2feslint-plugin)
updater | 2024/10/03 12:06:18 INFO <job_update_0_npm_and_yarn_all> Requirements to unlock own
updater | 2024/10/03 12:06:18 INFO <job_update_0_npm_and_yarn_all> Requirements update strategy bump_versions
updater | 2024/10/03 12:06:18 INFO <job_update_0_npm_and_yarn_all> Updating eslint from 7.32.0 to 9.11.1
##[debug]Agent environment resources - Disk: / Available 19061.47 MB out of 74244.74 MB, Memory: Used 1246.00 MB out of 6930.00 MB, CPU: Usage 58.02%
  proxy | 2024/10/03 12:06:21 [266] GET [https://registry.npmjs.org:443/@angular-eslint%2feslint-plugin-template](https://registry.npmjs.org/@angular-eslint%2feslint-plugin-template)
  proxy | 2024/10/03 12:06:21 [266] 200 [https://registry.npmjs.org:443/@angular-eslint%2feslint-plugin-template](https://registry.npmjs.org/@angular-eslint%2feslint-plugin-template)
  proxy | 2024/10/03 12:06:22 [271] GET [https://registry.npmjs.org:443/@angular-eslint%2fbundled-angular-compiler](https://registry.npmjs.org/@angular-eslint%2fbundled-angular-compiler)
  proxy | 2024/10/03 12:06:22 [272] GET [https://registry.npmjs.org:443/aria-query](https://registry.npmjs.org/aria-query)
  proxy | 2024/10/03 12:06:22 [273] GET [https://registry.npmjs.org:443/axobject-query](https://registry.npmjs.org/axobject-query)
  proxy | 2024/10/03 12:06:22 [274] GET [https://registry.npmjs.org:443/@angular-eslint%2futils](https://registry.npmjs.org/@angular-eslint%2futils)
  proxy | 2024/10/03 12:06:22 [272] 200 [https://registry.npmjs.org:443/aria-query](https://registry.npmjs.org/aria-query)
  proxy | 2024/10/03 12:06:22 [273] 200 [https://registry.npmjs.org:443/axobject-query](https://registry.npmjs.org/axobject-query)
  proxy | 2024/10/03 12:06:22 [271] 200 [https://registry.npmjs.org:443/@angular-eslint%2fbundled-angular-compiler](https://registry.npmjs.org/@angular-eslint%2fbundled-angular-compiler)
  proxy | 2024/10/03 12:06:22 [274] 200 [https://registry.npmjs.org:443/@angular-eslint%2futils](https://registry.npmjs.org/@angular-eslint%2futils)
##[debug]Agent environment resources - Disk: / Available 19058.73 MB out of 74244.74 MB, Memory: Used 1270.00 MB out of 6930.00 MB, CPU: Usage 100.00%
##[debug]Agent environment resources - Disk: / Available 19053.45 MB out of 74244.74 MB, Memory: Used 1314.00 MB out of 6930.00 MB, CPU: Usage 54.34%
##[debug]Agent environment resources - Disk: / Available 19052.84 MB out of 74244.74 MB, Memory: Used 1352.00 MB out of 6930.00 MB, CPU: Usage 53.87%
##[debug]Agent environment resources - Disk: / Available 19052.25 MB out of 74244.74 MB, Memory: Used 1397.00 MB out of 6930.00 MB, CPU: Usage 52.79%
##[debug]Agent environment resources - Disk: / Available 19051.68 MB out of 74244.74 MB, Memory: Used 1419.00 MB out of 6930.00 MB, CPU: Usage 52.84%
##[debug]Agent environment resources - Disk: / Available 19050.57 MB out of 74244.74 MB, Memory: Used 1437.00 MB out of 6930.00 MB, CPU: Usage 54.08%
  proxy | 2024/10/03 12:06:58 Skipping sending metrics because api endpoint is empty
##[debug]Agent environment resources - Disk: / Available 19049.98 MB out of 74244.74 MB, Memory: Used 1454.00 MB out of 6930.00 MB, CPU: Usage 54.08%
##[debug]Agent environment resources - Disk: / Available 19049.52 MB out of 74244.74 MB, Memory: Used 1488.00 MB out of 6930.00 MB, CPU: Usage 52.58%
##[debug]Agent environment resources - Disk: / Available 19048.99 MB out of 74244.74 MB, Memory: Used 1487.00 MB out of 6930.00 MB, CPU: Usage 52.55%

**Reply by rhyskoedijk** [reply, marked as answer]
@jasonddown just an update on this; I've got Task V2 security-only updates mostly working in https://github.com/tinglesoftware/dependabot-azure-devops/pull/1394, it shouldn't be too far away now. Hopefully manually update eslint got you past this issue? If not, it might be worth while searching the [dependabot-core project](https://github.com/dependabot/dependabot-core/issues) to see if there are any existing issues similar to this, or log an issue if not.

**Reply by jasonddown** [reply]
It has, I've manually updated eslint to the latest version and it went past that point.
I did try to update eslint to a more recent version but not the latest and it would still hang.
